lib/elixir/lib/enum.ex

```diff
@@ -2259,14 +2266,12 @@ defmodule Enum do
     do: []
   def take_random(first..first, count) when is_integer(count) and count >= 1,
     do: [first]
-  def take_random(first..last, 1) when first > last,
-    do: take_random(last..first, 1)
-  def take_random(first..last, 1),
-    do: [random_index(last - first) + first]
+  def take_random(_.._ = range, 1),
+    do: [random(range)]
```

Comment:
the code you mentioned above worked, for the 1.2 branch,
in 1.3 there are two commits of my own that changed the semantics, which are: 76d175a0fb44e460c1699e7eaed6a8acc7113cf5 and 2d707e25f70d66e39eff8ff0a25c9a1dc5306019

let me know how should we proceed.
I'm so sorry about this, didn't think of this at all.

We should definitely add your example as a test, and also one dealing with an empty enumerable